libtenzir/builtins/aggregation-functions/max.cpp

```diff
@@ -42,7 +49,82 @@ class max_function final : public aggregation_function {
   std::optional<type_to_data_t<Type>> max_ = {};
 };
 
-class plugin : public virtual aggregation_function_plugin {
+class max_instance final : public aggregation_instance {
+public:
+  explicit max_instance(ast::expression expr) : expr_{std::move(expr)} {
+  }
+
+  auto update(const table_slice& input, session ctx) -> void override {
+    auto arg = eval(expr_, input, ctx);
+    // TODO: Collapse them into a generic, or think about compute api
+    auto f = detail::overload{
+      [](const arrow::NullArray&) {},
+      [&](const arrow::Int64Array& array) {
+        for (auto i = int64_t{}; i < array.length(); ++i) {
+          if (array.IsValid(i)) {
+            const auto val = array.Value(i);
+            if (not max_ || val > std::get<int64_t>(max_.value())) {
+              max_ = val;
+            }
+          }
+        }
+      },
+      [&](const arrow::UInt64Array& array) {
+        for (auto i = int64_t{}; i < array.length(); ++i) {
+          if (array.IsValid(i)) {
+            const auto val = array.Value(i);
+            if (not max_ || val > std::get<uint64_t>(max_.value())) {
+              max_ = val;
+            }
+          }
+        }
+      },
+      [&](const arrow::DoubleArray& array) {
+        for (auto i = int64_t{}; i < array.length(); ++i) {
+          if (array.IsValid(i)) {
+            const auto val = array.Value(i);
+            if (not max_ || val > std::get<double>(max_.value())) {
+              max_ = val;
+            }
+          }
+        }
+      },
```

Comment:
As we discussed in the huddle, this needs 
1. no unchecked `std::get`
2. checked conversion to handle mixed numeric types (maybe refactored into an abstraction)
3. preferably not repeat itself that much